crates/starknet/src/blocks/mod.rs

```diff
@@ -54,12 +59,35 @@ impl StarknetBlocks {
         self.num_to_block.insert(block_number, block);
         self.last_block_hash = Some(hash);
     }
+
+    pub fn connect_state_diff_to_block(
+        &mut self,
+        block_number: BlockNumber,
+        state_diff: StateDiff,
+    ) {
+        self.num_to_state_diff.insert(block_number, state_diff);
+    }
+
+    pub fn get_by_block_id(&self, block_id: BlockId) -> Option<&StarknetBlock> {
+        match block_id {
+            BlockId::Hash(hash) => self.get_by_hash(Felt::from(hash)),
+            BlockId::Number(block_number) => self.num_to_block.get(&BlockNumber(block_number)),
+            BlockId::Tag(BlockTag::Latest) => {
+                if let Some(hash) = self.last_block_hash {
```

Comment:
you have solved the problem I had in my PR here :D

Reply:
Sorry for not letting you know about it, you had spend your time going through the PR